vms/platformvm/state/state.go

```diff
@@ -1827,26 +1829,29 @@ type subnetNodePair struct {
 	nodeID   ids.NodeID
 }
 
-type validatorStatusPair struct {
-	validator *Staker
-	status    diffValidatorStatus
+type stakerStatusPair struct {
+	staker *Staker
+	status diffValidatorStatus
 }
 
 type results struct {
-	weightDiffs map[subnetNodePair]*ValidatorWeightDiff
-	blsKeyDiffs map[ids.NodeID]*bls.PublicKey
-	valSetDiff  map[subnetNodePair]validatorStatusPair
+	modifiedCurrentStakers []stakerStatusPair
+	weightDiffs            map[subnetNodePair]*ValidatorWeightDiff
+	blsKeyDiffs            map[ids.NodeID]*bls.PublicKey
+	valSetDiff             map[subnetNodePair]stakerStatusPair
 }
 
 func (s *state) processCurrentStakers() (results, error) {
 	var (
+		outputStakers = make([]stakerStatusPair, 0)
 		outputWeights = make(map[subnetNodePair]*ValidatorWeightDiff)
 		outputBlsKey  = make(map[ids.NodeID]*bls.PublicKey)
-		outputValSet  = make(map[subnetNodePair]validatorStatusPair)
+		outputValSet  = make(map[subnetNodePair]stakerStatusPair)
 	)
 
 	for subnetID, subnetValidatorDiffs := range s.currentStakers.validatorDiffs {
-		// for now, let writeCurrentStakers consume s.currentStakers.validatorDiffs
+		delete(s.currentStakers.validatorDiffs, subnetID)
```

Comment:
I don't think it really matter. I keep the delete as we go since we used to do that in writeCurretStakers. But we could switch to wiping the map after the loop too.